client/src/components/list-of-universities/ListOfUniversities.js

```diff
@@ -1,22 +1,60 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
 
 import { Input } from '../common';
 import UniversityCard from '../university-card';
 
-const ListOfUniversities = ({ data }) => {
+import Spinner from '../spinner';
+import ErrorIndicator from '../error-indicator';
+
+import TestService from '../../service/TestService';
+import { useRequest } from '../hooks';
+
+const ListOfUniversities = ({ filters }) => {
+  const { getUniversities } = new TestService();
```

Comment:
you can use context to inject your service

Reply:
resolved